### PR TITLE
Atmospherics Envirohelmets now have the same fire protection as the Atmospherics Hardsuit Helmet

### PR DESCRIPTION
**Pull request by FernandoJ8**
## About The Pull Request

This PR makes the Atmospherics Plasmamen Envirohelmet have the same max_heat_protection_temperature as the Atmospheric Hardsuit helmet. The description of the Atmospherics Envirohelmet has also been changed to reflect this, now having the same "Has thermal shielding." line at the end as the Atmospheric Hardsuit and Hardsuit helmet.

## Why It's Good For The Game

If you have a Plasmamen Envirohelmet 99% of the time you have easy access to an Atmospherics Hardsuit, so this is simply for the convenience of not having to take off your envirohelmet to get the full fire protection of wearing a Hardsuit.
 

## Changelog
:cl:
qol: Atmospherics envirohelmet has the same fire protection as the Atmospherics hardsuit helmet.  It's description has also changed to reflect this.
/:cl:

### CODE REVIEW

**Comment by GuillaumePrata** [comment]
Removing your plasmaman helmet is always bad because people don't realize you're a plasmaman and fully strip you at medbay so a change like this is helpful.

Just a small thing, if Atmos plasmaman are getting this, so should the CE.

**Comment by FernandoJ8** [comment]
> Just a small thing, if Atmos plasmaman are getting this, so should the CE.

Very true, on it.

File: code/modules/clothing/spacesuits/plasmamen.dm [hunk 1]
```diff
@@ -225,17 +225,22 @@
 
 /obj/item/clothing/head/helmet/space/plasmaman/atmospherics
 	name = "atmospherics plasma envirosuit helmet"
-	desc = "A space-worthy helmet specially designed for atmos technician plasmamen, the usual purple stripes being replaced by atmos' blue."
+	desc = "A space-worthy helmet specially designed for atmos technician plasmamen, the usual purple stripes being replaced by atmos' blue. Has thermal shielding." // Added the thermal shielding clarification
 	icon_state = "atmos_envirohelm"
 	inhand_icon_state = "atmos_envirohelm"
 	armor = list(MELEE = 0, BULLET = 0, LASER = 0, ENERGY = 0, BOMB = 0, BIO = 100, RAD = 10, FIRE = 100, ACID = 75)
+	heat_protection = HEAD
```
Comment (GuillaumePrata) [inline code/modules/clothing/spacesuits/plasmamen.dm:232]: This line isn't needed, the normal space helmets have it already.
```suggestion
```

File: code/modules/clothing/spacesuits/plasmamen.dm [hunk 2]
```diff
@@ -225,17 +225,22 @@
...
+	max_heat_protection_temperature = FIRE_IMMUNITY_MAX_TEMP_PROTECT // Same protection as the Atmospherics Hardsuit helmet
 
 /obj/item/clothing/head/helmet/space/plasmaman/chief_engineer
 	name = "chief engineer's plasma envirosuit helmet"
-	desc = "A special containment helmet designed for the Chief Engineer, the usual purple stripes being replaced by the chief's green."
+	desc = "A special containment helmet designed for the Chief Engineer, the usual purple stripes being replaced by the chief's green. Shines with a high polish." // Added the same polish line that the CE hardsuit has
 	icon_state = "ce_envirohelm"
 	inhand_icon_state = "ce_envirohelm"
 	armor = list(MELEE = 0, BULLET = 0, LASER = 0, ENERGY = 0, BOMB = 0, BIO = 100, RAD = 10, FIRE = 100, ACID = 75)
+	heat_protection = HEAD
```
Comment (GuillaumePrata) [inline code/modules/clothing/spacesuits/plasmamen.dm:241]: Same here.
```suggestion
```

File: code/modules/clothing/spacesuits/plasmamen.dm [hunk 3]
```diff
@@ -225,17 +225,22 @@
...
 	icon_state = "atmos_envirohelm"
 	inhand_icon_state = "atmos_envirohelm"
 	armor = list(MELEE = 0, BULLET = 0, LASER = 0, ENERGY = 0, BOMB = 0, BIO = 100, RAD = 10, FIRE = 100, ACID = 75)
+	heat_protection = HEAD
+	max_heat_protection_temperature = FIRE_IMMUNITY_MAX_TEMP_PROTECT // Same protection as the Atmospherics Hardsuit helmet
 
 /obj/item/clothing/head/helmet/space/plasmaman/chief_engineer
 	name = "chief engineer's plasma envirosuit helmet"
-	desc = "A special containment helmet designed for the Chief Engineer, the usual purple stripes being replaced by the chief's green."
+	desc = "A special containment helmet designed for the Chief Engineer, the usual purple stripes being replaced by the chief's green. Shines with a high polish." // Added the same polish line that the CE hardsuit has
```
Comment (GuillaumePrata) [inline code/modules/clothing/spacesuits/plasmamen.dm:237]:
```suggestion
	desc = "A special containment helmet designed for the Chief Engineer, the usual purple stripes being replaced by the chief's green. Has improved thermal shielding."
```
Thermal shielding is clear on how it works to the players, more than the polish line.

File: code/modules/clothing/spacesuits/plasmamen.dm [hunk 4]
```diff
@@ -225,17 +225,22 @@
 
 /obj/item/clothing/head/helmet/space/plasmaman/atmospherics
 	name = "atmospherics plasma envirosuit helmet"
-	desc = "A space-worthy helmet specially designed for atmos technician plasmamen, the usual purple stripes being replaced by atmos' blue."
+	desc = "A space-worthy helmet specially designed for atmos technician plasmamen, the usual purple stripes being replaced by atmos' blue. Has thermal shielding." // Added the thermal shielding clarification
```
Comment (GuillaumePrata) [inline code/modules/clothing/spacesuits/plasmamen.dm:228]:
```suggestion
	desc = "A space-worthy helmet specially designed for atmos technician plasmamen, the usual purple stripes being replaced by atmos' blue. Has improved thermal shielding." // Added the thermal shielding clarification
```
The normal space suits already have thermal shielding but not the atmos helmet level.

**Review by GuillaumePrata — COMMENTED** [review]
Some small details to make it cleaner.

File: code/modules/clothing/spacesuits/plasmamen.dm (hunk 1 above)
Comment (FernandoJ8) [reply code/modules/clothing/spacesuits/plasmamen.dm:232]: Ah, missed that. Thank you for pointing it out

File: code/modules/clothing/spacesuits/plasmamen.dm (hunk 4 above)
Comment (FernandoJ8) [reply code/modules/clothing/spacesuits/plasmamen.dm:228]: I mean, yeah, but the other space suits don't have a "Has thermal shielding line" like the atmos one has. I think it's better to keep it like this for the sake of consistency with other descriptions

File: code/modules/clothing/spacesuits/plasmamen.dm (hunk 3 above)
Comment (FernandoJ8) [reply code/modules/clothing/spacesuits/plasmamen.dm:236]: It was mostly to be consistent with the description of the CE's Hardsuit, but yeah, it's clearer to change it to that.

File: code/modules/clothing/spacesuits/plasmamen.dm (hunk 4 above)
Comment (GuillaumePrata) [reply code/modules/clothing/spacesuits/plasmamen.dm:228]: Yes but having improved or another word to point out it is as good as the hardsuit halmet helps the players. Helmets do have a little line on examine on how much fire protection they give tho so either works.

Comment (FernandoJ8) [reply code/modules/clothing/spacesuits/plasmamen.dm:228]: Eh, I changed my mind about that, your proposal's better.

**Comment by LemonInTheDark** [comment]
I'm not sure I like making plasmamen even better at engineering/atmos then they already are, especially since there's typically only one atmos hardsuit per station.

File: code/modules/clothing/spacesuits/plasmamen.dm [hunk 5]
```diff
@@ -225,17 +225,20 @@
 
 /obj/item/clothing/head/helmet/space/plasmaman/atmospherics
 	name = "atmospherics plasma envirosuit helmet"
-	desc = "A space-worthy helmet specially designed for atmos technician plasmamen, the usual purple stripes being replaced by atmos' blue."
+	desc = "A space-worthy helmet specially designed for atmos technician plasmamen, the usual purple stripes being replaced by atmos' blue. Has improved thermal shielding." // Added the thermal shielding clarification
```
Comment (LemonInTheDark) [inline code/modules/clothing/spacesuits/plasmamen.dm:228]: Don't make comments that describe what your pr does, they're for people in the future, not now.

File: code/modules/clothing/spacesuits/plasmamen.dm [hunk 6]
```diff
@@ -225,17 +225,20 @@
...
 	icon_state = "atmos_envirohelm"
 	inhand_icon_state = "atmos_envirohelm"
 	armor = list(MELEE = 0, BULLET = 0, LASER = 0, ENERGY = 0, BOMB = 0, BIO = 100, RAD = 10, FIRE = 100, ACID = 75)
+	max_heat_protection_temperature = FIRE_IMMUNITY_MAX_TEMP_PROTECT // Same protection as the Atmospherics hardsuit Helmet
```
Comment (LemonInTheDark) [inline code/modules/clothing/spacesuits/plasmamen.dm:232]: These are fine, since they describe why it's the way it is

**Review by LemonInTheDark — CHANGES_REQUESTED** [review]

**Comment by GuillaumePrata** [comment]
> I'm not sure I like making plasmamen even better at engineering/atmos then they already are, especially since there's typically only one atmos hardsuit per station.

It will mostly affect the mini gamble of "will they realize I'm a plasmaman" at medbay since people don't realize you're a plasmaman at glance if you don't have your standard uniform and will strip you.
Atmos suit + atmos helmet, if they remove either you burn and die
Hardsuit + hardsuit helmet, if they remove either you burn and die
With this you can keep the helmet for clarity but still is limited by the number of hardsuits in the round since a fire protection helmet alone does nothing.

**Comment by kriskog** [comment]
Seems ok to me, atmos techs are supposed to be sufficiently equipped to fight fires.

**Pull request merged**